mava/components/jax/building/distributor.py

```diff
@@ -54,26 +57,27 @@ def on_building_program_nodes(self, builder: SystemBuilder) -> None:
             nodes_on_gpu=self.config.nodes_on_gpu,
             name=self.config.distributor_name,
             terminal=self.config.terminal,
+            lp_launch_type=self.config.lp_launch_type,
         )
 
         # tables node
         data_server = builder.store.program.add(
-            builder.data_server,
+            copy_store(builder, self.config.multi_process).data_server,
```

Comment:
I think we should rather do this (see likewise elsewhere) in the `add` method of `launcher.py`. There's already a check there for single/multi-processing.

This can be done by either passing through the builder as an argument as well, or perhaps by accessing `self` from the `node_fn`. Not sure if the latter is possible, but worth a try; if not, the former will still be quite neat.

The copy should happen just before the following is run:
```process = node_fn(*arguments)```

Reply:
Agreed :slightly_smiling_face:  Option one is definitely possible and if it looks cleaner we should do it. I intially tried option two, but it seemed quite hard to drive the self, as you mentioned, from the function itself. Maybe it is possible?

Reply:
Hmmm, not actually sure how to do option one. Once the function is created it is already too late. Any ideas @mmorris44 ?

Reply:
Option 2 can be achieved with the following:
```
class Class:
    def __init__(self):
        self.param = 1
        
    def method(self):
        return self.param
    
obj = Class()
fn = obj.method
print(fn())
fn.__self__.param = 2
print(fn())
```

Reply:
But we can possibly use your method if we call the same function on the buildercopy and return that :+1:

Reply:
@OmaymaMahjoub figure it out :partying_face:

Reply:
Just need to fix the launcher unit test! And I think it will be done!

Reply:
Implemented small updates to get to option 1. Let me know if you are happy with it @OmaymaMahjoub :slightly_smiling_face:

Reply:
I will just run pre commit and check the tests to verify why the checks failed